armasec/token_security.py

```diff
@@ -80,19 +75,21 @@ async def __call__(self, request: Request) -> TokenPayload:
         is injected to a route endpoint via the Depends() method. Lazily loads the OIDC config,
         the TokenDecoder, and the TokenManager if they are not already initialized.
         """
-        if self.manager is None:
-            self.debug_logger("Lazy loading TokenManager")
-            loader = OpenidConfigLoader(self.domain, debug_logger=self.debug_logger)
-            decoder = TokenDecoder(loader.jwks, self.algorithm, debug_logger=self.debug_logger)
-            self.manager = TokenManager(
-                loader.config,
-                decoder,
-                audience=self.audience,
-                debug_logger=self.debug_logger,
-            )
+
+        token_payload = None
 
         try:
-            token_payload = self.manager.extract_token_payload(request.headers)
+            for domain_config in self.domains_config:
+                try:
+                    manager = self._retrieve_manager(domain_config)
+                    token_payload = manager.extract_token_payload(request.headers)
+                    self.manager = manager
+                    break  # it means there is a matching jwk
+                except AuthenticationError:
+                    self.debug_logger(f"Failed to match JWK against domain {domain_config.domain}")
+                except Exception as err:
+                    if self.debug_exceptions:
+                        self.debug_logger(f"Exception caught: {err.__class__.__name__}")
```

Comment:
Unexpected exceptions need to be re-raised here. This is a code path that we would not expect to be hit, and if it is, we need to do some troubleshooting. Thus, we don't want to suppress the error.

Additionally, you need to log more than just the exception class's name, otherwise you don't have enough information to go off of. It would be a good idea to log the stack trace in addition to the error message.

Reply:
this is from the previous implementation actually, I just copied :raised_eyebrow: